src/modules/routing/components/PrivateRoute.tsx

```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import { isEmpty } from 'lodash';
+import { useSelector } from 'react-redux';
+import { Route, Redirect, RouteProps } from 'react-router-dom';
+
+import { ApplicationState } from 'modules/redux-store';
+import { getUserData } from 'modules/user';
+
+const PrivateRoute = (props: RouteProps) => {
+  const user = useSelector((state: ApplicationState) => getUserData()(state));
```

Comment:
This can actually be written much shorter.

` useSelector(getUserData);`

When passing no arguments other then state, pass by ref is the cleaner choice imho.

https://react-redux.js.org/next/api/hooks#using-memoizing-selectors